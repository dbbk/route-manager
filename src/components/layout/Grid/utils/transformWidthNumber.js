export default function transformWidthNumber(w: number) {
  return w >= 0 && w <= 1 ? `${w * 100}%` : `${w}px`;
}
