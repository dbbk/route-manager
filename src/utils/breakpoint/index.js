/* eslint consistent-return: 0 default-case: 0 */
/* stylelint-disable */

import { css } from 'styled-components';
import theme from '../theme';

export default function breakpoint(label: string) {
  return (...args: any) => () => {
    switch (label) {
      case 'xs':
        return css`
          @media (max-width: ${theme.breakpoints.xs}px) {
            ${css(...args)};
          }
        `;

      case 'sm':
        return css`
          @media (min-width: ${theme.breakpoints.xs}px) and (max-width: ${theme
              .breakpoints.sm}px) {
            ${css(...args)};
          }
        `;

      case 'md':
        return css`
          @media (min-width: ${theme.breakpoints.sm}px) and (max-width: ${theme
              .breakpoints.md}px) {
            ${css(...args)};
          }
        `;

      case 'lg':
        return css`
          @media (min-width: ${theme.breakpoints.md}px) {
            ${css(...args)};
          }
        `;

      case 'to-sm':
        return css`
          @media (max-width: ${theme.breakpoints.sm}px) {
            ${css(...args)};
          }
        `;

      case 'to-md':
        return css`
          @media (max-width: ${theme.breakpoints.md}px) {
            ${css(...args)};
          }
        `;

      case 'to-lg':
        return css`
          @media (max-width: ${theme.breakpoints.lg}px) {
            ${css(...args)};
          }
        `;

      case 'from-xs':
        return css`
          @media (min-width: ${theme.breakpoints.xs}px) {
            ${css(...args)};
          }
        `;

      case 'from-sm':
        return css`
          @media (min-width: ${theme.breakpoints.sm}px) {
            ${css(...args)};
          }
        `;

      case 'from-md':
        return css`
          @media (min-width: ${theme.breakpoints.md}px) {
            ${css(...args)};
          }
        `;

      case 'from-lg':
        return css`
          @media (min-width: ${theme.breakpoints.lg}px) {
            ${css(...args)};
          }
        `;

      case 'from-xlg':
        return css`
          @media (min-width: ${theme.breakpoints.xlg}px) {
            ${css(...args)};
          }
        `;
    }
  };
}
