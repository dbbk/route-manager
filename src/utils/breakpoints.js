// @flow

// Breakpoints are defined as a map of (name: minimum width)
// https://zpl.io/25RJPjV
const breakpoints = {
  xs: 0, // officially we support a minimum width of 320px
  sm: 569,
  md: 769,
  lg: 961,
  xlg: 1221, // DEPRECATED
};

export default breakpoints;
