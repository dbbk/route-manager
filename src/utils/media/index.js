// @flow

import bp from 'src/utils/breakpoints';

export default function media(label: string) {
  return () => {
    switch (label) {
      case 'xs':
        return `@media (max-width: ${bp.sm - 1}px)`;

      case 'sm':
        return `@media (min-width: ${bp.sm}px) and (max-width: ${bp.md - 1}px)`;

      case 'md':
        return `@media (min-width: ${bp.md}px) and (max-width: ${bp.lg - 1}px)`;

      case 'lg':
        return `@media (min-width: ${bp.lg}px)`;

      case 'to-sm':
        return `@media (max-width: ${bp.sm - 1}px)`;

      case 'to-md':
        return `@media (max-width: ${bp.md - 1}px)`;

      case 'to-lg':
        return `@media (max-width: ${bp.lg - 1}px)`;

      case 'from-sm':
        return `@media (min-width: ${bp.sm}px)`;

      case 'from-md':
        return `@media (min-width: ${bp.md}px)`;

      case 'from-lg':
        return `@media (min-width: ${bp.lg}px)`;

      default:
        return null;
    }
  };
}
