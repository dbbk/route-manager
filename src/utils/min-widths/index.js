// @flow

// minWidths defines the minimum widths of our responsive breakpoints
// as a map of (name: minimum width)
// https://zpl.io/25RJPjV
const minWidths = {
  xs: 0, // officially we support a minimum width of 320px
  sm: 569,
  md: 769,
  lg: 961,
  xlg: 1221, // DEPRECATED
};

export default minWidths;
