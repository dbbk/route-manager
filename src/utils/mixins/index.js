// @flow

import { css } from 'styled-components';
import { prop, switchProp, withProp } from 'styled-tools';
import { darken, lighten } from 'polished';

export const colorMixin = switchProp('color', {
  black: css`
    ${prop('theme.colors.greyscale.black')};
  `,
  grey: css`
    ${prop('theme.colors.greyscale.dark')};
  `,
  white: css`
    ${prop('theme.colors.white')};
  `,
  red: css`
    ${prop('theme.colors.danger')};
  `,
  primary: css`
    ${prop('theme.colors.brand.primary')};
  `,
  inherit: 'inherit',
});

// NOTE while we are lighening and darkening colours using withProp,
// "babel-plugin-polished" will not work.
export const hoverColorMixin = switchProp('color', {
  black: css`
    ${withProp('theme.colors.greyscale.black', lighten(0.1))};
  `,
  grey: css`
    ${withProp('theme.colors.greyscale.dark', darken(0.1))};
  `,
  white: css`
    ${withProp('theme.colors.white', darken(0.1))};
  `,
  red: css`
    ${withProp('theme.colors.danger', darken(0.1))};
  `,
  primary: css`
    ${withProp('theme.colors.brand.primary', darken(0.1))};
  `,
  inherit: 'inherit',
});
