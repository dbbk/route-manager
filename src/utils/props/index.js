// @flow

export function blacklistProps(props: {}, blacklist: Array<string> = []): {} {
  const newProps = { ...props };

  blacklist.forEach(blacklistedProp => delete newProps[blacklistedProp]);

  return newProps;
}
