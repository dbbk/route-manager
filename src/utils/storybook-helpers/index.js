// @flow
import styled from 'styled-components';

const BlackBackground = styled.div`
  background-color: black;
  padding: 1em;
`;

const WhiteBackground = styled.div`
  background-color: white;
  padding: 1em;
`;

const GreyBackground = styled.div`
  background-color: lightgrey;
  padding: 1em;
  text-align: center;
  font-family: monospace;
`;

const FullWidthBackground = styled.div`
  height: 100vh;
  width: 100vw;
`;

export const Background = {
  White: WhiteBackground,
  Black: BlackBackground,
  Grey: GreyBackground,
  FullWidth: FullWidthBackground,
};
