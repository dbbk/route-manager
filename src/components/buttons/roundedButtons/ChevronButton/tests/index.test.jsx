// @flow

import React from 'react';
import { ChevronButton } from '@festicket/react-ui-components';
import { renderWithTheme } from 'src/utils/testing';

describe('<ChevronButton />', () => {
  test('as button', () => {
    const tree = renderWithTheme(
      <ChevronButton element="button" direction="right" />,
    );
    expect(tree).toMatchSnapshot();
  });
  test('as anchor', () => {
    const tree = renderWithTheme(
      <ChevronButton element="a" direction="right" />,
    );
    expect(tree).toMatchSnapshot();
  });
});
