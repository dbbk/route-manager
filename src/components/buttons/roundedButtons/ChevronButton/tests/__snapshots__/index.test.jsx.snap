// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChevronButton /> as anchor 1`] = `
.c4 {
  height: 18px;
  fill: #A2A4AD;
}

.c1 {
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #8C64CB;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0 {
  display: inline-block;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 50%;
  padding: 0;
  cursor: pointer;
  background-color: #F5F5F5;
}

.c0:hover {
  background-color: rgba(36,37,41,0.1);
  color: #9497a1;
}

.c0 .c2 {
  width: 28px;
  height: 28px;
}

<a
  className="c0 c1"
  element="a"
  target="_blank"
  variant="standard"
  weight="bold"
>
  <div
    className="c2 c3"
  >
    <svg
      className="c4"
      color="grey"
      viewBox="0 0 18 18"
    >
      <path
        d="M5.8 3.3L11.4 9l-5.6 5.7 1.4 1.4 7-7.1-7-7.1z"
      />
    </svg>
  </div>
</a>
`;

exports[`<ChevronButton /> as button 1`] = `
.c4 {
  height: 18px;
  fill: #A2A4AD;
}

.c1 {
  border: none;
  outline: none;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0 {
  display: inline-block;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 50%;
  padding: 0;
  cursor: pointer;
  background-color: #F5F5F5;
}

.c0:hover {
  background-color: rgba(36,37,41,0.1);
  color: #9497a1;
}

.c0 .c2 {
  width: 28px;
  height: 28px;
}

<button
  aria-disabled={undefined}
  className="c0 c1"
  disabled={undefined}
  size="regular"
>
  <div
    className="c2 c3"
  >
    <svg
      className="c4"
      color="grey"
      viewBox="0 0 18 18"
    >
      <path
        d="M5.8 3.3L11.4 9l-5.6 5.7 1.4 1.4 7-7.1-7-7.1z"
      />
    </svg>
  </div>
</button>
`;
